Version Mismatch

Build is broken after dependabot's bump to `eslint v10` — `eslint-plugin-react-hooks@7.0.1` only supports up to `eslint 9`.

Either:
- Pin `eslint` back to `^9.0.0`
- Or upgrading `eslint-plugin-react-hooks` to a version that supports `eslint 10` (if one exists)

Build error message:

```text
> make build-vite-nc
Building vite without cache in dev mode...
WARN[0000] Docker Compose requires buildx plugin to be installed 
Sending build context to Docker daemon  21.24MB
Step 1/9 : FROM node:25-alpine
 ---> deb3acc1bf15
Step 2/9 : WORKDIR /app
 ---> Running in d31949acea3d
 ---> Removed intermediate container d31949acea3d
 ---> 0bf12d3e6218
Step 3/9 : COPY package*.json ./
 ---> 5a8dd905dccd
Step 4/9 : RUN npm install -g npm@latest
 ---> Running in a3eae45c6162

changed 19 packages in 4s

15 packages are looking for funding
  run `npm fund` for details
 ---> Removed intermediate container a3eae45c6162
 ---> 460d5b076c4d
Step 5/9 : RUN npm install
 ---> Running in 971de61671e9
npm error code ERESOLVE
npm error ERESOLVE could not resolve
npm error
npm error While resolving: eslint-plugin-react-hooks@7.0.1
npm error Found: eslint@10.0.2
npm error node_modules/eslint
npm error   dev eslint@"^10.0.2" from the root project
npm error   peer eslint@"^6.0.0 || ^7.0.0 || >=8.0.0" from @eslint-community/eslint-utils@4.9.1
npm error   node_modules/@eslint-community/eslint-utils
npm error     @eslint-community/eslint-utils@"^4.8.0" from eslint@10.0.2
npm error   2 more (@eslint/js, eslint-plugin-react-refresh)
npm error
npm error Could not resolve dependency:
npm error peer eslint@"^3.0.0 || ^4.0.0 || ^5.0.0 || ^6.0.0 || ^7.0.0 || ^8.0.0-0 || ^9.0.0" from eslint-plugin-react-hooks@7.0.1
npm error node_modules/eslint-plugin-react-hooks
npm error   dev eslint-plugin-react-hooks@"^7.0.1" from the root project
npm error
npm error Conflicting peer dependency: eslint@9.39.4
npm error node_modules/eslint
npm error   peer eslint@"^3.0.0 || ^4.0.0 || ^5.0.0 || ^6.0.0 || ^7.0.0 || ^8.0.0-0 || ^9.0.0" from eslint-plugin-react-hooks@7.0.1
npm error   node_modules/eslint-plugin-react-hooks
npm error     dev eslint-plugin-react-hooks@"^7.0.1" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry this command with --force or --legacy-peer-deps to accept an incorrect (and potentially broken) dependency resolution.
npm error
npm error
npm error For a full report see:
npm error /root/.npm/_logs/2026-03-12T10_14_01_226Z-eresolve-report.txt
npm error A complete log of this run can be found in: /root/.npm/_logs/2026-03-12T10_14_01_226Z-debug-0.log
[+] build 0/1
 ⠙ Image secure-drive-vite Building                                                                                                                     17.9s
The command '/bin/sh -c npm install' returned a non-zero code: 1
make: *** [Makefile:131: build-vite-nc] Error 1
```

--- a/frontend/package.json
+++ b/frontend/package.json
@@ -24,7 +24,7 @@
     "@types/react": "^19.2.14",
     "@types/react-dom": "^19.2.3",
     "@vitejs/plugin-react": "^5.1.4",
-    "eslint": "^10.0.2",
+    "eslint": "^9.0.0",
     "eslint-plugin-react-hooks": "^7.0.1",
     "eslint-plugin-react-refresh": "^0.5.2",
     "globals": "^17.4.0",

